### ISSUE TITLE
Altcover fails after partial migration + nuget updates into .net5

### ISSUE THREAD

**Issue by dalibor-sanzeru** [SteveGilham/altcover#125]
I have migrated partially (only EF stays on lower versions 2.*) solution into .net 5, but dotnet test with altcover fails with error:

`dotnet.exe test --configuration "Debug" /p:AltCover=true /p:AltCoverCobertura="\coverage.xml" /p:AltCoverAssemblyFilter="NUnit|PredavamApi.Tests"`

Not expected result:

```
 Determining projects to restore...
  All projects are up-to-date for restore.
  PredavamApi.Common -> C:\Workspace\Vyrabam\PredavamApi.Common\bin\Debug\netstandard2.0\PredavamApi.Common.dll
  PredavamApi.DataSeed -> C:\Workspace\Vyrabam\PredavamApi.DataSeed\bin\Debug\netstandard2.0\PredavamApi.DataSeed.dll
  PredavamApi -> C:\Workspace\Vyrabam\PredavamApi\bin\Debug\net5.0\PredavamApi.dll
  PredavamApi.Tests -> C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\PredavamApi.Tests.dll
  Instrumenting files from C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\
  Writing files to C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\__Instrumented_PredavamApi.Tests
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\BouncyCastle.Crypto.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\FluentValidation.AspNetCore.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\FluentValidation.DependencyInjectionExtensions.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\FluentValidation.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\Humanizer.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\Magick.NET-Q8-AnyCPU.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\Magick.NET.Core.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\Moq.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\MySqlConnector.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\PredavamApi.Common.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\PredavamApi.DataSeed.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\PredavamApi.dll
     => C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\System.Interactive.Async.dll

  Coverage Report: C:\Workspace\Vyrabam\PredavamApi.Tests\coverage.xml


      C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\__Instrumented_PredavamApi.Tests\System.Interactive.Async.dll
                  <=  System.Interactive.Async, Version=3.2.0.0, Culture=neutral, PublicKeyToken=null
      C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\__Instrumented_PredavamApi.Tests\PredavamApi.Common.dll
                  <=  PredavamApi.Common, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
      C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\__Instrumented_PredavamApi.Tests\MySqlConnector.dll
                  <=  MySqlConnector, Version=0.49.2.0, Culture=neutral, PublicKeyToken=null
      C:\Workspace\Vyrabam\PredavamApi.Tests\bin\Debug\net5.0\__Instrumented_PredavamApi.Tests\Moq.dll
                  <=  Moq, Version=4.16.0.0, Culture=neutral, PublicKeyToken=null
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018: The "AltCover.Prepare" task failed unexpectedly. [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object. [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Gendarme.I.switchCyclomaticComplexity@98-2.Invoke(Instruction previous) [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.FSharp.Core.OptionModule.Filter[T](FSharpFunc`2 predicate, FSharpOption`1 option) in D:\workspace\_work\1\s\src\fsharp\FSharp.Core\option.fs:line 74 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Gendarme.I.switchCyclomaticComplexity@71.Invoke(Int32 c, Instruction i) in C:\projects\altcover\AltCover.Engine\Gendarme.fs:line 95 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.FSharp.Collections.SeqModule.Fold[T,TState](FSharpFunc`2 folder, TState state, IEnumerable`1 source) in D:\workspace\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 731 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Gendarme.I.switchCyclomaticComplexity(IEnumerable`1 instructions) in C:\projects\altcover\AltCover.Engine\Gendarme.fs:line 107 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.OpenCover.methodElement@194(MethodDefinition methodDef) in C:\projects\altcover\AltCover.Engine\OpenCover.fs:line 195 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.OpenCover.visitMethod$cont@234(OpenCoverContext s, MethodDefinition methodDef, Inspections included, Unit unitVar) in C:\projects\altcover\AltCover.Engine\OpenCover.fs:line 235 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.OpenCover.visitMethod@231-1(OpenCoverContext s, MethodDefinition methodDef, Inspections included) in C:\projects\altcover\AltCover.Engine\OpenCover.fs:line 232 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.OpenCover.reportVisitor@666-1.Invoke(OpenCoverContext s, Node node) in C:\projects\altcover\AltCover.Engine\OpenCover.fs:line 670 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Visitor.stateful@1415-1.Invoke(T node) in C:\projects\altcover\AltCover.Engine\Visitor.fs:line 1417 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult](FSharpFunc`2 mapping, FSharpList`1 x) in D:\workspace\_work\1\s\src\fsharp\FSharp.Core\local.fs:line 247 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Visitor.visit@1405.Invoke(FSharpList`1 visitors, Node node) in C:\projects\altcover\AltCover.Engine\Visitor.fs:line 1405 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.FSharp.Collections.SeqModule.Fold[T,TState](FSharpFunc`2 folder, TState state, IEnumerable`1 source) in D:\workspace\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 731 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Visitor.visit(IEnumerable`1 visitors, IEnumerable`1 assemblies) in C:\projects\altcover\AltCover.Engine\Visitor.fs:line 1403 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Main.I.result@588.Invoke(Unit unitVar0) in C:\projects\altcover\AltCover.Engine\Main.fs:line 609 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.CommandLine.I.doPathOperation[a](FSharpFunc`2 f, a defaultValue, Boolean store) in C:\projects\altcover\AltCover.Engine\CommandLine.fs:line 233 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.CommandLine.doPathOperation@470.Invoke(FSharpFunc`2 f, a defaultValue, Boolean store) in C:\projects\altcover\AltCover.Engine\CommandLine.fs:line 470 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Main.I.doInstrumentation(String[] arguments) in C:\projects\altcover\AltCover.Engine\Main.fs:line 586 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Main.I.main(String[] arguments) in C:\projects\altcover\AltCover.Engine\Main.fs:line 663 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Main.effectiveMain@669.Invoke(String[] arguments) in C:\projects\altcover\AltCover.Engine\Main.fs:line 669 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at AltCover.Prepare.Execute() in C:\projects\altcover\AltCover.Engine\Tasks.fs:line 189 [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
C:\Users\dalib\.nuget\packages\altcover\8.1.819\build\netstandard2.0\AltCover.targets(83,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\Workspace\Vyrabam\PredavamApi.Tests\PredavamApi.Tests.csproj]
```

Thanks for any advice.

**Comment by SteveGilham** [comment]
While the report is enough to tell me what's going wrong and where is shall have to make a fix -- the method being inspected for its complexity while being instrumented has what appears to be a redundant branch expression, presumably from building at a new compiler version -- I can't say for sure which assembly the interesting code is in.

The balance of probability is that it's in one of the third-party assemblies not excluded from instrumentation; and if that is the case, excluding all third-party code from instrumentation may work around the problem while waiting for a proper fix.  Such assemblies could be excluded either by `/p:AltCoverLocalSource=true` to exclude assemblies referring to source code not on the current machine, or by explicitly naming the unwanted ones e.g. /p:AltCoverAssemblyFilter="?PredavamApi|Tests"  to exclude assemblies not having `PredavamApi` in their name, and ones that do have `Tests` ).

**Comment by dalibor-sanzeru** [comment]
Hi Steve,
Both of your proposed solutions worked for me.

Thank You.

**Comment by SteveGilham** [comment]
Should be addressed in release [v8.2.820](https://www.nuget.org/packages/altcover/8.2.820).
